server/src/controllers/apps.controller.ts

```diff
@@ -284,8 +284,12 @@ export class AppsController {
   @UseGuards(JwtAuthGuard)
   @Delete(':id/versions/:versionId')
   async deleteVersion(@User() user, @Param('id') id, @Param('versionId') versionId) {
+    const number = await this.appsService.fetchVersions(user, id);
     const version = await this.appsService.findVersion(versionId);
     const ability = await this.appsAbilityFactory.appsActions(user, id);
+    if (number.length <= 1) {
```

Comment:
Shall we add this after the ability checks?
https://github.com/ToolJet/ToolJet/pull/5123/files#diff-4cc88c767a6288a89c3d84175bac75d447c6fda36b23275ffcd88f323d2e979bR294

Reply:
I think that should be it now.